LDMP/models/datasets.py

```diff
@@ -90,26 +134,179 @@ def child(self, row: int) -> None:
+        self.datasetsStore[dump_file_name] = dataset
+
+        return (dump_file_name, dataset)
+
+    # method useful to interface with MVC model
     def columnCount(self) -> int:
         return 1
 
     def rowCount(self) -> int:
-        return len(self.datasets)
+        return len(self.datasetsStore)
     
     def child(self, row: int) -> Optional['Dataset']:
-        if row < 0 or row >= len(self.datasets):
+        if row < 0 or row >= len(self.datasetsStore):
             return None
-        return self.datasets[row]
+        return list(self.datasetsStore.items())[row][1]
     
     def columnName(self, column: int) -> Optional[str]:
         return None
+
+    def sync(self):
+        """Method to sync with Datasets available in "trends_earth/advanced/base_data_directory".
+
+        The method parse content o base_data_directory and create Dataset instance for each 
+        found Dataset descriptor.
+        """
+        base_data_directory = QSettings().value("trends_earth/advanced/base_data_directory", None, type=str)
+
+        def traverse(path):
+            excluded_path = os.path.sep + 'Jobs' + os.path.sep
+
+            for basepath, directories, files in os.walk(path):
+                for f in files:
+                    # skip files in Jobs
+                    if excluded_path.lower() in basepath.lower():
+                        continue
+                    
+                    yield os.path.join(basepath, f)
+
+        # remove any previous memorised Datasets
+        # self.reset()
+        schema = DatasetSchema()
+        for json_file in traverse(base_data_directory):
+            # skip larger thatn 1MB files
+            statinfo = os.stat(json_file)
+            if statinfo.st_size > 1024*1024:
+                continue
+
+            # skip any not json file
+            with open(json_file, 'r') as fd:
+                try:
+                    parsed_json = json.load(fd)
+                except json.decoder.JSONDecodeError as ex:
+                    # no json => skip
+                    continue
+                except Exception as ex:
+                    log(tr('Error reading file {} with ex: ').format(json_file, str(ex)))
+                    continue
+            
+                try:
+                    dataset_dict = schema.load(parsed_json, partial=True, unknown=marshmallow.INCLUDE)
+                    dataset = Dataset(dataset=dataset_dict)
+                    self.datasetsStore[json_file] = dataset
+
+                except marshmallow.exceptions.ValidationError as ex:
+                    log(tr('not a valid Dataset {} with ex: ').format(f, str(ex)))
+                    continue
+                except Exception as ex:
+                    # Exception notification managed in append method
+                    log(tr('Cannot manage dataset for file {} ex: {}').format(json_file, str(ex)))
+                    continue
+        self.updated.emit()
+
+
+class DatasetSchema(Schema):
```

Comment:
schema is located here and not in shema.py because schema.py belogs to a nested project and I do not want/permission to touch it